src/subscription/subscribe.js

```diff
@@ -106,16 +110,33 @@ function subscribeImpl(
   fieldResolver,
   subscribeFieldResolver
 ) {
-  const subscription = createSourceEventStream(
+  // If arguments are missing or incorrect, throw an early error.
+  assertValidExecutionArguments(
     schema,
     document,
-    rootValue,
-    contextValue,
-    variableValues,
-    operationName,
-    subscribeFieldResolver
+    variableValues
   );
 
+  let subscription;
+  try {
+    subscription = createSourceEventStream(
+      schema,
+      document,
+      rootValue,
+      contextValue,
+      variableValues,
+      operationName,
+      subscribeFieldResolver
+    );
+  } catch (error) {
+    // GraphQLError represent "field" or "query" errors, to be returned to
+    // the requesting client as a GraphQL Response. Other errors are "internal"
+    // errors, which cause the event stream to fail.
+    return error instanceof GraphQLError ?
```

Comment:
Error is a super-class of GraphQLError and typically covers everything else, but technically *anything* can be thrown, so what we want is to check if it is a GraphQLError or otherwise